src/odemis/driver/pmtctrl.py

```diff
@@ -447,6 +470,36 @@ def _sendCommand(self, cmd):
 
             return ans.rstrip()
 
+    def _tryRecover(self):
+        self._recovering = True
+        self.state._set_value(HwError("PMTControl disconnected"), force_write=True)
+        # Retry to open the serial port (in case it was unplugged)
+        # _ser_access should already be acquired, but since it's an RLock it can be acquired
+        # again in the same thread
+        with self._ser_access:
+            while True:
+                try:
+                    self._serial.close()
+                    self._serial = None
+                except Exception:
+                    pass
+                try:
+                    logging.debug("retrying to open port %s", self._port)
+                    self._port = self._findDevice(self._portname)
+                except IOError:
+                    time.sleep(2)
+                except Exception:
+                    logging.exception("Unexpected error while trying to recover device")
+                    raise
+                else:
+                    # We found it back!
+                    break
```

Comment:
Indeed, as the IOError actually continues (which I hadn't noticed), the break would need to go inside the try/except, which would be less readable. So the `else:` is good :-)